apps/journeys-admin/src/components/Editor/EditToolbar/Menu/TemplateSettingsDialog/TemplateSettingsDialog.tsx

```diff
@@ -74,7 +74,14 @@ export function TemplateSettingsDialog({
     featured: journey?.featuredAt != null,
     strategySlug: journey?.strategySlug,
     tagIds: journey?.tags.map(({ id }) => id),
-    creatorDescription: journey?.creatorDescription
+    creatorDescription: journey?.creatorDescription,
+    language: {
+      id: journey?.language?.id as string,
+      localName: journey?.language?.name.find(({ primary }) => !primary)
+        ?.value as string,
+      nativeName: journey?.language?.name.find(({ primary }) => primary)
+        ?.value as string
+    }
   }
```

Comment:
this needs to be languageId.
The underlying component should handle the logic of converting from languageId to language and then back to languageId. See how the tags selectors work.

```suggestion
    languageId: journey?.language?.id
```